addons/repair/static/tests/tours/repair_order.js

```diff
@@ -0,0 +1,52 @@
+/** @odoo-module */
+
+import { registry } from "@web/core/registry";
+
+registry.category("web_tour.tours").add('test_repair_without_product_in_parts', {
+
+    steps: () => [
+    {
+        content: "Click product_id field",
+        trigger: "div[name=move_ids] td[name=product_id].o_field_cell",
+        run: "click",
+    },
+    {
+        content: "Unset the product",
+        trigger: "div[name=move_ids] input.o-autocomplete--input",
+        run: function (el) {
+            el.anchor.value = '';
+            el.anchor.dispatchEvent(new Event("change", { bubbles: true }));
+        },
+    },
+    {
+        content: "Click partner field",
+        trigger: "div[name=partner_id] input.o-autocomplete--input",
+        run: "click",
+    },
+    // Note: Selecting the partner is only to trigger the compute;
+    // this could be done by modifying any other field.
+    {
+        content: "Select partner",
+        trigger: "ul.ui-autocomplete a:contains('A Partner')",
+        run: "click",
+    },
+    {
+        content: "Click the product field",
+        trigger: "div[name=move_ids] input.o-autocomplete--input",
+        run: "click",
+    },
+    {
+        content: "Select a product",
+        trigger:"ul.o-autocomplete--dropdown-menu li.o-autocomplete--dropdown-item.ui-menu-item > a:contains('[1234] A Product')",
+        run: "click",
+    },
+    {
+        content: "Save the repair order",
+        trigger: ".fa-cloud-upload",
```

Comment:
use more specific/clearer classes

```suggestion
        trigger: ".o_form_button_save",
```